File: examples/typescript/cdktf/templates/s3-stack.ts
import { Stack, StackGroup } from '@awslv/cdktf-organizer';
import { AwsProvider } from '@cdktf/provider-aws';
import { S3Bucket } from '@cdktf/provider-aws/lib/s3';
import { Construct } from 'constructs';

export type S3StackProps = {
  bucketName: string;
};

export class S3Stack extends Stack {
  public bucket: S3Bucket;

  constructor(
    stackGroup: StackGroup,
    scope: Construct,
    id: string,
    props: S3StackProps
  ) {
    super(stackGroup, scope, id);

    new AwsProvider(this, 'Aws', {
      region: this.config['region'] as string,
    })

    this.bucket = new S3Bucket(this, 'Bucket', {
      bucket: this.getBucketName(props.bucketName)
    });
  }
}

File: examples/python/cdktf/templates/s3_stack.py
"""S3 Bucket Stack."""

from dataclasses import dataclass
from typing import TYPE_CHECKING

from cdktf_cdktf_provider_aws.provider import AwsProvider
from cdktf_cdktf_provider_aws.s3_bucket import S3Bucket

if TYPE_CHECKING:
    from constructs import Construct
    from cdk_organizer.terraform.stack_group import StackGroup

from cdk_organizer.terraform.stack import Stack


@dataclass
class S3BucketStackParameters:
    base_bucket_name: str


class S3BucketStack(Stack):
    def __init__(
        self,
        scope: "Construct",
        id: str,
        stack_group: "StackGroup",
        parameters: S3BucketStackParameters,
        **kwargs
    ) -> None:
        super().__init__(scope, id, stack_group, **kwargs)

        AwsProvider(
            self,
            'Aws',
            region=self.config.get('region'),
        )

        self.bucket = S3Bucket(
            self,
            'S3Bucket',
            bucket=self.get_bucket_name(parameters.base_bucket_name)
        )
